test/io-tests/test_io.py

```diff
@@ -226,17 +232,41 @@ def test_iat_claim():
     token = jwt.encode(claim, secret).decode("utf-8")
     headers = {"Authorization": f"Bearer {token}"}
 
-    with run(basedir / "configs" / "simple.config") as url:
+    with run(basedir / "configs" / "simple.config") as process:
         for _ in range(10):
-            response = requests.get(f"{url}/authors_only", headers=headers)
+            response = requests.get(f"{process.baseurl}/authors_only", headers=headers)
             assert response.status_code == 200
 
             time.sleep(0.5)
 
 
 def test_app_settings():
-    with run(basedir / "configs" / "app-settings.config") as baseurl:
-        url = f"{baseurl}/rpc/get_guc_value?name=app.settings.external_api_secret"
+    with run(basedir / "configs" / "app-settings.config") as process:
+        url = (
+            f"{process.baseurl}/rpc/get_guc_value?name=app.settings.external_api_secret"
+        )
         response = requests.get(url)
         assert response.status_code == 200
         assert response.text == '"0123456789abcdef"'
+
+
+def test_app_settings_reload(tmp_path):
+    config = (basedir / "configs" / "sigusr2-settings.config").read_text()
+    configfile = tmp_path / "test.config"
+    configfile.write_text(config)
+
+    with run(configfile) as process:
+        url = f"{process.baseurl}/rpc/get_guc_value?name=app.settings.name_var"
+
+        response = requests.get(url)
+        assert response.status_code == 200
+        assert response.text == '"John"'
+
+        # change setting
+        configfile.write_text(config.replace("John", "Jane"))
+        # reload
+        process.process.send_signal(signal.SIGUSR2)
```

Comment:
`process.process` :D

Maybe rename `PostgrestProcess` to just `Postgrest` and `as process` to `as postgrest`. Then it would be `postgrest.process` and `postgrest.baseurl` ?

Reply:
You're reviewing faster than I can code :-) Yes, I'm trying to see what is needed and how it fits together, in the end several pieces will be simplified/streamlined!